### ISSUE TITLE
Uploaded Video thumbnail not display at all.

### ISSUE THREAD

**Issue by StellarKuldeep** [smilefam/SendBird-iOS#58]
Recently I have written the code to Video upload in chat and I use SendBird demo project code to upload Video in chat, but the issue is Thumbnail not display in uploaded video.

I try this in SendBird demo as well as in my project, but facing the same issue in both.

I also follow this Docs to generate the thumbnail from video url but no luck. 
https://docs.sendbird.com/ios/group_channel_advanced

I use this like below.

```
// success, data is in imageData
/***********************************/
/* Thumbnail is a premium feature. */
/***********************************/
let thumbnailSize = SBDThumbnailSize.make(withMaxWidth: 320.0, maxHeight: 320.0)
                    
let preSendMessage = self.groupChannel.sendFileMessage(withBinaryData: (videoFileData! as Data), filename: (videoName as String), type: mimeType, size: UInt((videoFileData?.length)!), thumbnailSizes: [thumbnailSize!], data: "", customType: "", progressHandler: nil, completionHandler: { (fileMessage, error) in
    //My Code
}
```

Once the video uploaded, Video plays successfully.

Please help me regarding this.
Thanks in advance

**Comment by sf-jed-kyung** [comment]
Hi, @StellarKuldeep 

Let me know the App ID in AppDelegate.swift of SendBird demo app that you see. 
_**Do not post your App ID here.**_

Thanks.

**Comment by StellarKuldeep** [comment]
@sf-jed-kyung , Thank for your reply. My App ID of SendBird demo app is **9DA1B1F4-0BE6-4DA8-82C5-2E81DAB56F23**

**Comment by famous90** [comment]
> @sf-jed-kyung , Thank for your reply. My App ID of SendBird demo app is **9DA1B1F4-0BE6-4DA8-82C5-2E81DAB56F23**

**Comment by StellarKuldeep** [comment]
@famous90 May i know why you close this question?

**Comment by famous90** [comment]
@StellarKuldeep sorry to miss close.

**Comment by sf-jed-kyung** [comment]
@StellarKuldeep The thumbnail feature is enabled on the demo app. I think the filename and the mime type cause the problem. Let me know the values of them.

**Comment by StellarKuldeep** [comment]
@sf-jed-kyung  Filename is **8BD990A9-76EF-45C9-94D5-343194FCD4E8.MOV** and mimetype is  **video/quicktime**

**Comment by StellarKuldeep** [comment]
@sf-jed-kyung Today i am not able send text messages, Image/video file message.

When i send any message to any one then it shows me status as sending but when i go back and come again then image/video file message is not there.

Please find the attached screenshot, as you can see the file message status is sending.

![Simulator Screen Shot - iPhone 8 - 2019-03-20 at 15 03 02](https://user-images.githubusercontent.com/38131385/54674106-bdb0df00-4b21-11e9-9ef8-cd706c32307a.png)


When i try to Resend message then i'm getting an error like below screen.

![Simulator Screen Shot - iPhone 8 - 2019-03-20 at 15 07 51](https://user-images.githubusercontent.com/38131385/54674250-023c7a80-4b22-11e9-8a3b-aa3c1bc87baa.png)

In above screenshots, There's no video thumbnail available.

Please help me regarding this.

**Comment by sf-jed-kyung** [comment]
@StellarKuldeep  I'm really sorry. The permission option of the demo app was wrong and I've changed it. Could you try it again?

**Comment by StellarKuldeep** [comment]
@sf-Jed-Kyung, Thanks, Text messages, chat now works fine but still i am facing video thumbnail problem. Please help me regarding this.

**Comment by sf-jed-kyung** [comment]
@StellarKuldeep The filename and the mime type are normal.  The debugging is not easy now. So, can you create a group channel and invite `sendbird_test_jed` user to the channel on the demo app? If you send a video in the channel, then I'll take a look into the thumbnail issue.

**Comment by StellarKuldeep** [comment]
@sf-jed-kyung , I have created 1 group channel and invite you `sendbird_test_jed` and  i was able to send 2 text messages and then i faced below error.

![Simulator Screen Shot - iPhone 8 - 2019-03-26 at 17 51 26](https://user-images.githubusercontent.com/38131385/54997187-2d770c00-4ff1-11e9-938a-a3795f19da76.png)

I already send video in this channel and I attached a screenshot. It shows me only "SENDING" and once i go back from this channel and come back to this channel, the sent video was gone.

![Simulator Screen Shot - iPhone 8 - 2019-03-26 at 18 00 01](https://user-images.githubusercontent.com/38131385/54997317-7202a780-4ff1-11e9-9058-a2c16a11816d.png)


Please help me about this.

**Comment by sf-jed-kyung** [comment]
@StellarKuldeep I turned off the profanity filter. Can you try to upload a video again? I'm watching that channel.

**Comment by StellarKuldeep** [comment]
@sf-jed-kyung  few minutes ago I uploaded the video in created channel but video not sent successfully and still thumbnail doesn't generate.

**Comment by sf-jed-kyung** [comment]
![IMG_0025 2](https://user-images.githubusercontent.com/5144488/55379517-345ebc80-54d2-11e9-8ac2-237100fd5351.PNG)

@StellarKuldeep 
I can see the thumbnails.

Could you compare your implementation with [this](https://github.com/smilefam/SendBird-iOS/blob/master/basic/old/sample-swift/SendBird-iOS/Chatting/ViewCell/OutgoingVideoFileMessageTableViewCell.swift#L66-L75)?

**Comment by StellarKuldeep** [comment]
@sf-Jed-Kyung yes, the highlighted code exists in both `IncomingVideoFileMessageTableViewCell` & `outgoingVideoFileMessageTableViewCell`

**Comment by sf-jed-kyung** [comment]
@StellarKuldeep I tested the project of this repo without any modification.

**Comment by StellarKuldeep** [comment]
@sf-jed-kyung , yes, but i have been facing this issue in the newly downloaded demo. Will you please try to send video from your end and check weather video thumbnail shows from your end or not?

**Comment by abdulKarim002** [comment]
@StellarKuldeep same happens to me also I have cross-checked all the code and still  no video preview on chat

**Comment by StellarKuldeep** [comment]
@abdulKarim002 , Still video preview isn't generating. If you found any solution then please let me know.

**Comment by erickimme** [comment]
@StellarKuldeep Can you take a look at this [how to generate thumbnail](https://docs.google.com/document/d/1eZfL7QXGzqkcrgarFCHunVO8MPXpGaKeRe5GVEm2ju0/edit?usp=sharing) guide?

**Comment by StellarKuldeep** [comment]
@erickimme , It not open source guide, it asks me to request access.

**Comment by erickimme** [comment]
@StellarKuldeep  Hi, I just send you an email with how to generate thumbnail guide since I can't attache a file here.

**Comment by StellarKuldeep** [comment]
@erickimme , I have tried that code but what should I pass in below code?

```
self.chattingView.preSendFileData[preSendMessage?.requestId!] = [
                    "data": videoFileData as AnyObject,
                    "type": mimeType as AnyObject,
                ]
self.chattingView.preSendMessages[preSendMessage.requestId!] = preSendMessage
self.chattingView.messages.append(preSendMessage)
self.chattingView.chattingTableView.reloadData()
DispatchQueue.main.async {
    self.chattingView.scrollToBottom(force: true)
}
```
